### ISSUE TITLE
ITagQuery doesn't perform as expected with multiple cultures

### ISSUE THREAD

**Issue by matthewcare** [umbraco/Umbraco-CMS#15707]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.1.0 - 13.2.0-rc.preview.33

### Bug summary

I'm not 100% if this is the intended behaviour, however it was certainly very confusing to me, and definitely doesn't feel right.

Tags are not returned correctly from ITagQuery if a tag property is vary by culture.
If a tag is marked as vary by culture then you _must_ pass in a culture when using ITagQuery in order to get these tags, however then tags not marked as vary by culture are not returned.

### Specifics

Content types with tag properties that are a variant won't be included in ITagQuery data unless a culture is specified, however if you specify a culture then non-variant tag properties are not included.

![image](https://github.com/umbraco/Umbraco-CMS/assets/1469061/49412cf8-5907-47ec-afe0-a9bebcf60bcd)

```
@inject ITagQuery TagQuery

....

// Only returns non-variant tags
var tags = TagQuery.GetAllContentTags();
foreach (var tag in tags)
{
    <p>@tag?.Text</p>
}

// Only returns variant tags for en-US
var enTags = TagQuery.GetAllContentTags(culture: "en-US");
foreach (var tag in enTags)
{
    <p>@tag?.Text</p>
}

// Only returns variant tags for sv
var svTags = TagQuery.GetAllContentTags(culture: "sv");
foreach (var tag in svTags)
{
    <p>@tag?.Text</p>
}
```

### Steps to reproduce

- Create a content type that is "Allow vary by culture".
- Create two tag properties, and allow one of them to vary by culture
- Implement the provided code
- Observe that the variant tags do not show unless a culture is passed in
- Observe that the non-variant tags do not show when a culture is passed in

### Expected result / actual result

Expected:
`TagQuery.GetAllContentTags()` should return tags that are varying by culture, as well as tags that are not varying by culture.
`TagQuery.GetAllContentTags(culture: "en-US")` should return all tags that are not varying by culture as well as tags that are varying by culture.

Actual:
`TagQuery.GetAllContentTags()` only returns non-varying tags
`TagQuery.GetAllContentTags(culture: "en-US")` only return varying tags

A work around is to perform both queries, and join the results. Since this is probably what anyone that has come across this is already doing, changing the behaviour is going to be a breaking change :(

**Comment by andr317c** [comment]
Hey! 
I just took a look at your issue. When you are working with multiple cultures and want to get all tags, you will need to use "*".
`var tags = TagQuery.GetAllContentTags(culture: "*");`
Please let me know if this works

**Comment by matthewcare** [comment]
Hi @andr317c,

The problem is getting all tags for a single culture, not all cultures.
As I say, when getting tags for a single culture you must create two queries, one with no culture passed to get all tags which _are not_ marked as a variant, and then another query with the current culture passed in to get all tags which _are_ marked as a variant.

Passing in a wildcard will return all tags for all cultures.

To try to explain a little more.

### Swedish version of the page
- Variant tags property: En, två, tre
- Non variant tags property (shared with both versions): 1, 2, 3

### English version of the page
- Variant tags property: One, two, three
- Non variant tags property (shared with both versions): 1, 2, 3 

If I view the Swedish version of the page
`TagQuery.GetAllContentTags()`
will only return "1, 2, 3"

`TagQuery.GetAllContentTags(culture: "sv")`
will only return "En, två, tre"

`TagQuery.GetAllContentTags(culture: "*")`
will return "En, två, tre, One, two, three, 1, 2, 3"

I think this is incorrect. Querying for tags with no culture specified should still return tags for the current culture, as well as the non-culture specific tags.
Perhaps this is intended functionality though, and it's expected that you need to make two queries to get all of the tags for the current culture when you have both non-culture specific, and culture specific tag properties on one content type?

Thanks,
Matthew

**Comment by andr317c** [comment]
I agree that the document in the code is lacking, as it does not mention anything with invariant cultures if the culture is null. I will set this issue as an up for grabs. 
What should be fixed is the documentation in the file: `src/Umbraco.Core/PublishedCache/ITagQuery.cs`. The code summaries should specify that if you do not insert a culture, it gets tags with invariant culture, otherwise it will only get the specified culture, the '*' can be used to get all cultures.

**Comment by ageroh** [comment]
Hi all, 
I am looking in this issue and I think I am going to get my hands dirty, is this still needed to be fixed right?
Thank you

**Comment by korbi551** [comment]
This issue should be closed as it is already fixed in PR #15860

**Comment by AndyButland** [comment]
Closing as per the above comment and given the resolution was considered to be limited to updating the remarks on the method name.

I just pushed a [commit to tidy up the XML header documentation](https://github.com/umbraco/Umbraco-CMS/commit/313417cb91b04b253f81e56050626ee2eb3e1e69) - there was a double-comment in there from the original update in the linked PR, which perhaps interferes with how it's surfaced in intellisense.
